tools/ardusub.py

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/python
+
+from pymavlink import mavutil
+import time
+
+#Establish connection with pixhawk
+master = mavutil.mavlink_connection('udpout:localhost:9000')
+
+for x in range(3):
+	#Send request for autopilot version
+	master.mav.autopilot_version_request_send(
+		master.target_system,
+		master.target_component
+	)
+	msg = master.recv_match(type='AUTOPILOT_VERSION')
+
+	#Check if version information was obtained
+	#Print version
+	if msg:
+		version_d = msg.to_dict()
+		version_hex = hex(version_d["flight_sw_version"])
+		print ('.'.join(version for version in version_hex[2:-2:2]))
+		break
+
+	#Exit script after 3 attempts
+	if (x==2):
```

Comment:
This if is not necessary, you can just print error and return 1 after the for loop. Because it'll finish after the 3rd time.